src/lib/hooks.ts

```diff
@@ -29,3 +32,71 @@ export const useNotificationFromLocation = () => {
     }, []);
     return locationState;
 };
+
+type ApiResult<U> = RemoteData.RemoteData<ApiError | undefined, U>;
+// function useDispatchCall<T, U>(
+//     fn: AsyncThunk<U, T, { rejectValue: ApiError }>
+// ): [ApiResult<U>, (args: T, onSuccess?: (result: U) => void) => void] {
+//     const [apiResult, setApiResult] = useState<ApiResult<U>>(RemoteData.initial);
+//     const dispatch = useAppDispatch();
+
+//     const callFn = React.useCallback(
+//         (args: T, onSuccess?: (result: U) => void) => {
+//             if (!RemoteData.isPending(apiResult)) {
+//                 setApiResult(RemoteData.pending);
+
+//                 dispatch(fn(args)).then((action) => {
+//                     if (fn.fulfilled.match(action)) {
+//                         setApiResult(RemoteData.success(action.payload));
+//                         onSuccess?.(action.payload);
+//                     } else {
+//                         setApiResult(RemoteData.failure(action.payload));
+//                     }
+//                 });
+//             }
+//         },
+//         [apiResult]
+//     );
+
+//     return [apiResult, callFn];
+// }
+
+export function useApiCall<T, U>(
+    fn: (req: T) => Promise<ApiClientResult<U>>
+): [ApiResult<U>, (args: T, onSuccess?: (result: U) => void) => void] {
+    const [apiResult, setApiResult] = useState<ApiResult<U>>(RemoteData.initial);
+
+    const callFn = React.useCallback(
+        async (args: T, onSuccess?: (result: U) => void) => {
+            if (!RemoteData.isPending(apiResult)) {
+                setApiResult(RemoteData.pending);
+
+                const res = await fn(args);
+                if (res.status === 'ok') {
+                    setApiResult(RemoteData.success(res.data));
+                    onSuccess?.(res.data);
+                } else {
+                    setApiResult(RemoteData.failure(res.error));
+                }
+            }
+        },
+        [apiResult]
+    );
+
+    return [apiResult, callFn];
+}
+
+export function useFetchBrev<T>(
+    sliceFn: (args: T) => Promise<ApiClientResult<Blob, string>>
+): [ApiResult<Blob>, (args: T) => void] {
+    const [status, fetchBrev] = useApiCall(sliceFn);
+
+    return [
+        status,
+        (args: T) => {
+            fetchBrev(args, (blob) => {
+                window.open(URL.createObjectURL(blob));
+            });
+        },
+    ];
+}
```

Comment:
Denne gjør jo litt mer enn å bare hente det, så da kan det være greit å ha med det i navngivingen.

```suggestion
export function useBrevForhåndsvisning<T>(
    fetchBrev: (args: T) => Promise<ApiClientResult<Blob, string>>
): [ApiResult<Blob>, (args: T) => void] {
    const [status, forhåndsvisBrev] = useApiCall(fetchBrev);

    return [
        status,
        (args: T) => {
            forhåndsvisBrev(args, (blob) => {
                window.open(URL.createObjectURL(blob));
            });
        },
    ];
}
```